hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -423,6 +423,21 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		})
 	}
 
+	// Validate KubeVirt platform specific format
+	if nodePool.Spec.Platform.Type == hyperv1.KubevirtPlatform {
+		err := kubevirtPlatformValidation(nodePool)
+		if err != nil {
+			setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+				Type:               hyperv1.NodePoolValidKubevirtConfigConditionType,
+				Status:             corev1.ConditionFalse,
+				Reason:             hyperv1.NodePoolValidationFailedConditionReason,
+				Message:            fmt.Sprintf("validation of NodePool KubeVirt platform failed: %s", err.Error()),
+				ObservedGeneration: nodePool.Generation,
+			})
+			return ctrl.Result{}, fmt.Errorf("validation of NodePool KubeVirt platform failed: %w", err)
+		}
+	}
+
```

Comment:
We need to clean the condition here in case you go from failed to succeed, e.g removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolValidKubevirtConfigConditionType)

Reply:
ah, good catch, fixed!